poc_iot_injector/src/receipt_txn.rs

```diff
@@ -0,0 +1,244 @@
+use crate::{keypair::Keypair, Error, Result};
+use helium_crypto::Sign;
+use helium_proto::{
+    services::poc_lora::{
+        LoraBeaconReportReqV1, LoraValidBeaconReportV1, LoraValidPocV1, LoraValidWitnessReportV1,
+        LoraWitnessReportReqV1,
+    },
+    BlockchainPocPathElementV1, BlockchainPocReceiptV1, BlockchainPocWitnessV1,
+    BlockchainTxnPocReceiptsV2, Message,
+};
+use rust_decimal::{prelude::ToPrimitive, Decimal, MathematicalOps};
+use rust_decimal_macros::dec;
+use sha2::{Digest, Sha256};
+
+// TODO: These should ideally be coming in from some configuration service
+const WITNESS_REDUNDANCY: u32 = 4;
+const POC_REWARD_DECAY_RATE: Decimal = dec!(0.8);
+const HIP15_TX_REWARD_UNIT_CAP: Decimal = dec!(2.0);
+
+// TODO: This should be done better?
+const MHZTOHZ: u32 = 1000000;
+
+type PocPath = Vec<BlockchainPocPathElementV1>;
+
+pub fn handle_report_msg(
+    msg: prost::bytes::BytesMut,
+    keypair: &Keypair,
+    timestamp: i64,
+) -> Result<Option<(BlockchainTxnPocReceiptsV2, Vec<u8>, String)>> {
+    // NOTE: This is intentionally not Vec::with_capacity(1) since we _may_ add multi-hop PocPath
```

Comment:
couldn't we just convert this to `vec![]` if and when we do that? presumably we'd need more code changes than that and it wouldn't be as simple as toggling configuration?

Reply:
It _is_ `vec![]`

Reply:
i meant why not make it `Vec::with_capacity(1)` now and convert it to a `vec![]` if and when we do enable multi-hop paths

Reply:
Updated 16cbb24